### PR TITLE
Change `dirhash` package to `checksumdir`

### PR DESCRIPTION
**Pull request by martinlyra**
# TL;DR
This is a proposed fix to a compatibility issue with Windows as described in this issue: https://github.com/flyteorg/flyte/issues/1561 (with additional details in the Slack thread). In short the issue stems from an package called `dirtree` which depends on an other package by the same author; `scantree`. Since the packages are no longer maintained, `checksumdir` is proposed to replace it.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
The package `dirhash` which depends on the culprit `scantree` package is replaced by `checksumdir` in `fast_registration.py`

Unfortunately, `scantree` uses `posix` directly instead of `os` which becomes an issue with Windows. After somewhat thorough deliberation in Slack, we reached an impasse that we should replace `dirhash` with `checksumdir` and see if it would help resolve the compatibility issue.

In order to improve compatibility on Windows. An explicit `encoding="utf-8"` is added to an `open()` in `setup.py` as well.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1561

## Follow-up issue
https://github.com/flyteorg/flyte/issues/1561

### CODE REVIEW

**Comment by samhita-alla** [comment]
@martinlyra, thanks for creating this PR. Can we have `dirhash` removed and `checksumdir` added to the requirements.txt, dev-requirements.txt, doc-requirements.txt, setup.py and requirements-spark2.txt files?

cc: @wild-endeavor
